internal/characters/travelerdendro/burst.go

```diff
@@ -0,0 +1,82 @@
+package travelerdendro
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+)
+
+var burstFrames [][]int
+
+const burstHitmark = 91
+const leaLotusAppear = 54
+
+func init() {
+	burstFrames = make([][]int, 2)
+
+	// Male
+	burstFrames[0] = frames.InitAbilSlice(58)
+	burstFrames[0][action.ActionSwap] = 57 // Q -> Swap
+
+	// Female
+	burstFrames[1] = frames.InitAbilSlice(58)
+	burstFrames[1][action.ActionSwap] = 57 // Q -> Swap
+}
+
+func (c *char) Burst(p map[string]int) action.ActionInfo {
+
+	c.SetCD(action.ActionBurst, 1200)
+	c.ConsumeEnergy(2)
+
+	// Duration counts from first hitmark
```

Comment:
I'd move this calculation to lealotus.go instead of passing it a duration. the places that use burstDur can just use s.Gadget.Duration

Reply:
The gadget kills itself once it transfigs (it stops being targetable, stops reacting, etc, after transfig). So you can't just use s.Gadget.Duration I think.

Reply:
Hmm, burstDur is only used for burstExp and setting the gadget's duration. burstExp is pretty much only used for logging and queuing up tasks on burst activation so I don't see why you couldn't use s.Gadget.Duration there